bigtable-hbase-integration-tests/src/test/java/com/google/cloud/bigtable/hbase/TestBatch.java

```diff
@@ -366,4 +375,12 @@ public void testBatchGets() throws Exception {
 
     table.close();
   }
+
+  public void testBatchDoesntHang() throws Exception {
+    Connection closedConnection = ConnectionFactory.createConnection(IntegrationTests.getConfiguration());
+    Table table = closedConnection.getTable(TABLE_NAME);
+    closedConnection.close();
+    thrown.expect(RetriesExhaustedWithDetailsException.class);
```

Comment:
The reason I prefer checking the cause (even it is IllegalStateException) is that we set up a precondition that will eventually force the code to throw some exception, we want to to verify the exception is thrown from those async calls not somewhere else.

For example:

batch{
throw e1; // unexpected exception that could be caught in our test code.
try {
  async  aclls
} catch {
  throw e2; // expected exception to be caught in our test code.
}

} // batch

If you believe checking IllegalStateException could be confusing, it is something we have fix, not a reason for us not to check it.